File: Semestre 4/IASA_Aux/IASA/2022_2023/iasa45741/iasa-agente/src/lib/pee/mod/problema/problema.py
from abc import abstractmethod

class Problema():
    def __init__(self, estado_inicial, operadores):
        self._estado_inicial = estado_inicial #Estado inicial do problema. Lido através de um getter
        self._operadores = operadores # Operadores que transformam um estado no outro. Lido atravẽs de um getter
        
    @property
    def estado_inicial(self):
        return self._estado_inicial

    @property
    def operadores(self):
        return self._operadores

    @abstractmethod
    def objectivo(self, estado):
        # Recebe um estado e vê se é o estado final
        pass




# Como executar um problema:
# 1 - Definir um conjunto de operadores (Estado 1  ➤ Estado 2 com custo)
# 2 - Definir om estado inicial
# 3 - Definir um objetivo 
# 4 - Criar um problema passando operadores e estado inicial
# 5 - Executar o mecanismo de procura utilizado o problema criado

File: Semestre 4/IASA_Aux/IASA/2020_2021/tp2/src/tp2/modprob/Problema.java
package tp2.modprob;

public abstract class Problema {
	private Operador[] operadores;
	private Estado estadoInicial;
	public Problema(Estado estadoInicial, Operador[] operadores) {
		this.estadoInicial = estadoInicial;
		this.operadores = operadores;
	}
	
	public Operador[] getOperadores() {
		return operadores;
	}
	
	public Estado getEstadoInicial() {
		return estadoInicial;
		
	}
	
	public abstract boolean objetivo(Estado estado);

}
